examples/go-tc-counter/bpf/tc_counter.c

```diff
@@ -0,0 +1,57 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+#include <linux/bpf.h>
+#include <linux/pkt_cls.h>
+
+#include <bpf/bpf_helpers.h>
+
+// This counting program example was adapted from
+// https://github.com/xdp-project/xdp-tutorial/tree/master/basic03-map-counter
+
+/* This is the data record stored in the map */
+struct datarec {
+	__u64 rx_packets;
+	__u64 rx_bytes;
+} datarec;
+
+struct {
+	__uint(type, BPF_MAP_TYPE_PERCPU_ARRAY);
+	__type(key, __u32);
+	__type(value, datarec);
+	__uint(max_entries, TC_ACT_VALUE_MAX);
+	__uint(pinning, LIBBPF_PIN_BY_NAME);
+} tc_stats_map SEC(".maps");
+
+static __always_inline __u32 tc_stats_record_action(struct __sk_buff *skb,
```

Comment:
I don't recall what we did in the XDP variant, but `__always_inline` isn't required.

Reply:
It's the same in the xdp variant, but I can remove it.

Reply:
Done